### ISSUE TITLE
Delete user needs to delete forwards too

### ISSUE THREAD

**Issue by mtissington** [postfixadmin/postfixadmin#799]
I created a user and then created a single forward to that user.
I then deleted the user - the forward, now invalid, was not deleted.
The forward should have been deleted.

(could easily be implemented in sql with cascading delete).

**Comment by DavidGoodwin** [comment]
Hi @mtissington - thanks for the bug report - I think for historical reasons PostfixAdmin doesn't rely on database cascade rules (perhaps Sqlite doesn't support it?)

I assume an SQL delete needs to be added somewhere ...

**Comment by cboltz** [comment]
I'm afraid it isn't that easy. Things that at least need a second thought are (assuming you delete `user@example.com`)
* an alias `foo@example.com -> user@example.com, otheruser@example.com`
* an alias `foo@bar -> user@example.com` if a catchall `*@example.com` exists (in this case, the alias will end up in the catchall, which is IMHO better than not having the alias)
* and maybe more not-so-simple cases I can't imagine right now

To sum it up - I doubt that automatically deleting aliases is a good idea.

It might be an option to show a list of affected aliases after deleting a mailbox (shouldn't be too hard with the right list.php search parameters). Would that make sense?

**Comment by mtissington** [comment]
The alias would only be deleted when the number of valid destinations drops to zero.

Offering to delete the affected aliases would be a great idea :) at least showing a list.

**Comment by ozgurkazancci** [comment]
+1 Offering to delete/showing the affected aliases would be pretty useful.
